### ISSUE TITLE
AccessViolationException when disposing the VlcPlayer

### ISSUE THREAD

**Issue by Sayyid-A** [higankanshi/xZune.Vlc#83]
First off, thank you for such great VLC library. It works just fine, I have one issue though.

An AccessViolationException encountered when disposing the VlcPlayer in the window's OnClosing event, it occurs randomly while playing a stream or after the VlcMediaPlayer's EncounteredError event.
```
System.AccessViolationException: Attempted to read or write protected memory. This is often an indication that other memory is corrupt.

   at xZune.Vlc.VlcMediaPlayer.get_State()
   at xZune.Vlc.Wpf.VlcPlayer.<>c.<get_State>b__152_0(VlcMediaPlayer x)
   at xZune.Vlc.Wpf.Extension.DefaultValueWhenNull[T,TRet](T value, Func`2 selector, TRet defaultValue)
   at xZune.Vlc.Wpf.VlcPlayer.get_State()
   at xZune.Vlc.Wpf.VlcPlayer.<VlcMediaPlayerStateChanged>b__105_0()
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.WrappedInvoke(Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   at System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at MS.Internal.Threading.ExceptionFilterHelper.TryCatchWhen(Object source, Delegate method, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.WrappedInvoke(Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   at System.Windows.Threading.Dispatcher.Invoke(DispatcherPriority priority, Delegate method, Object arg)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   at MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.TranslateAndDispatchMessage(MSG& msg)
   at System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   at System.Windows.Threading.Dispatcher.Run()
   at System.Windows.Application.RunDispatcher(Object ignore)
   at System.Windows.Application.RunInternal(Window window)
   at System.Windows.Application.Run(Window window)
   at System.Windows.Application.Run()
```
I tried many workarounds and even to catch the exception using the `[HandleProcessCorruptedStateExceptions]` attribute, but it does not work!

**Comment by higankanshi** [comment]
I have tried to fix it, you can try the new code.

**Comment by Sayyid-A** [comment]
Thanks a lot  for your quick response!
I just found a workaround by stopping the VlcPlayer before disposing.

I'll try the new code soon and feedback.

**Comment by Sayyid-A** [comment]
Perfect! no exceptions so far with the last code.
Please update NuGet packages. Thanks again.

**Comment by higankanshi** [comment]
I want to fix and update it with #80 .
